addons/account/models/account_move.py

```diff
@@ -1528,18 +1528,26 @@ def _compute_quick_encoding_vals(self):
         for move in self:
             move.quick_encoding_vals = move._get_quick_edit_suggestions()
 
-    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date')
+    @api.depends('ref', 'move_type', 'partner_id', 'invoice_date', 'amount_total')
     def _compute_duplicated_ref_ids(self):
-        move_to_duplicate_move = self._fetch_duplicate_supplier_reference()
+        if all(move.is_sale_document() for move in self):
+            move_to_duplicate_move = self._fetch_duplicate_reference(move_type='out_invoice')
+        else:
+            move_to_duplicate_move = self._fetch_duplicate_reference(move_type='in_invoice')
         for move in self:
             move.duplicated_ref_ids = move_to_duplicate_move.get(move, self.env['account.move'])
 
-    def _fetch_duplicate_supplier_reference(self, only_posted=False):
-        moves = self.filtered(lambda m: m.is_purchase_document() and m.ref)
+    def _fetch_duplicate_reference(self, move_type, only_posted=False):
+        if move_type == 'out_invoice':
```

Comment:
Maybe we should consider credit note too?
At the moment, I think it also works for credit note but the button is just not displayed.
It might be worth to activate it for every invoice/refund type document.